tools/extras/install_clif.sh

```diff
@@ -0,0 +1,204 @@
+  echo "BUILD_DIR: $BUILD_DIR"
+  echo "PROTOBUF_INCLUDE: $PROTOBUF_INCLUDE"
+  echo "PROTOBUF_LIBS: $PROTOBUF_LIBS"
+  echo "CMAKE_PY_FLAGS: ${CMAKE_PY_FLAGS[@]}"
+  echo ""
+#  exit 1
+fi
+
+
+if [ -d "$CLIFSRC_DIR" ]; then
+  echo "Destination $CLIFSRC_DIR already exists!, skipping."
+  exit 0
+fi
+
+
+# Install clif from dogan's fork
+git clone $CLIF_GIT $CLIFSRC_DIR
+cd "$CLIFSRC_DIR"
+
+# If Ninja is installed, use it instead of make.  MUCH faster.
+
+declare -a CMAKE_G_FLAG
+declare -a MAKE_PARALLELISM
+if which ninja; then
+  CMAKE_G_FLAGS=(-G Ninja)
+  MAKE_OR_NINJA="ninja"
+  MAKE_PARALLELISM=()  # Ninja does this on its own.
+  # ninja can run a dozen huge ld processes at once during install without
+  # this flag... grinding a 12 core workstation with "only" 32GiB to a halt.
+  # linking and installing should be I/O bound anyways.
+  MAKE_INSTALL_PARALLELISM=(-j 2)
+  echo "Using ninja for the clif backend build."
+else
+  CMAKE_G_FLAGS=()  # The default generates a Makefile.
+  MAKE_OR_NINJA="make"
+  MAKE_PARALLELISM=(-j 2)
+  if [[ -r /proc/cpuinfo ]]; then
+    N_CPUS="$(cat /proc/cpuinfo | grep -c ^processor)"
+    [[ "$N_CPUS" -gt 0 ]] && MAKE_PARALLELISM=(-j $N_CPUS)
+    MAKE_INSTALL_PARALLELISM=(${MAKE_PARALLELISM[@]})
+  fi
+  echo "Using make.  Build will take a long time.  Consider installing ninja."
+fi
+
+# Download, build and install LLVM and Clang (needs a specific revision).
+
+mkdir -p "$LLVM_DIR"
+cd "$LLVM_DIR"
+svn co https://llvm.org/svn/llvm-project/llvm/trunk@307315 llvm
+cd llvm/tools
+svn co https://llvm.org/svn/llvm-project/cfe/trunk@307315 clang
+ln -s -f -n "$CLIFSRC_DIR/clif" clif
+
+# Build and install the CLIF backend.  Our backend is part of the llvm build.
+# NOTE: To speed up, we build only for X86. If you need it for a different
+# arch, change it to your arch, or just remove the =X86 line below.
+
+mkdir -p "$BUILD_DIR"
+cd "$BUILD_DIR"
+cmake -DCMAKE_INSTALL_PREFIX="$CLIF_VIRTUALENV/clang" \
+      -DCMAKE_PREFIX_PATH="$PROTOBUF_PREFIX_PATH" \
+      -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=true \
+      -DLLVM_INSTALL_TOOLCHAIN_ONLY=true \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DLLVM_BUILD_DOCS=false \
+      -DLLVM_TARGETS_TO_BUILD=X86 \
+      "${CMAKE_PY_FLAGS[@]}" \
+      "${CMAKE_G_FLAGS[@]}" "$LLVM_DIR/llvm"
+"$MAKE_OR_NINJA" "${MAKE_PARALLELISM[@]}" clif-matcher clif_python_utils_proto_util
+"$MAKE_OR_NINJA" "${MAKE_INSTALL_PARALLELISM[@]}" install
+
+# Get back to the CLIF Python directory and have pip run setup.py.
+
+cd "$CLIFSRC_DIR"
+# Grab the python compiled .proto
+cp "$BUILD_DIR/tools/clif/protos/ast_pb2.py" clif/protos/
+# Grab CLIF generated wrapper implementation for proto_util.
+cp "$BUILD_DIR/tools/clif/python/utils/proto_util.cc" clif/python/utils/
+cp "$BUILD_DIR/tools/clif/python/utils/proto_util.h" clif/python/utils/
+cp "$BUILD_DIR/tools/clif/python/utils/proto_util.init.cc" clif/python/utils/
+CFLAGS="$PROTOBUF_INCLUDE" LDFLAGS="$PROTOBUF_LIBS" "$PYTHON_PIP" install .
```

Comment:
do we need to set these environment variables for installing the pyclif package?